Verify public interface

Hi again,

In trying to use the new release in https://github.com/jeanp413/open-remote-ssh, there are several TypeScript errors due to how it is currently consumed by that package. [This is the relevant file](https://github.com/jeanp413/open-remote-ssh/blob/master/src/ssh/sshConfig.ts).

```
src/ssh/sshConfig.ts:4:21 - error TS2614: Module '"ssh-config"' has no exported member 'Directive'. Did you mean to use 'import Directive from "ssh-config"' instead?

4 import SSHConfig, { Directive, Line, Section } from 'ssh-config';
                      ~~~~~~~~~

src/ssh/sshConfig.ts:4:32 - error TS2614: Module '"ssh-config"' has no exported member 'Line'. Did you mean to use 'import Line from "ssh-config"' instead?

4 import SSHConfig, { Directive, Line, Section } from 'ssh-config';
                                 ~~~~

src/ssh/sshConfig.ts:4:38 - error TS2614: Module '"ssh-config"' has no exported member 'Section'. Did you mean to use 'import Section from "ssh-config"' instead?

4 import SSHConfig, { Directive, Line, Section } from 'ssh-config';
                                       ~~~~~~~

src/ssh/sshConfig.ts:49:37 - error TS2749: 'SSHConfig' refers to a value, but is being used as a type here. Did you mean 'typeof SSHConfig'?

49 function normalizeSSHConfig(config: SSHConfig) {
                                       ~~~~~~~~~

src/ssh/sshConfig.ts:68:45 - error TS2749: 'SSHConfig' refers to a value, but is being used as a type here. Did you mean 'typeof SSHConfig'?

68     const includedConfigs = new Map<number, SSHConfig>();
                                               ~~~~~~~~~

src/ssh/sshConfig.ts:97:36 - error TS2749: 'SSHConfig' refers to a value, but is being used as a type here. Did you mean 'typeof SSHConfig'?

97     constructor(private sshConfig: SSHConfig) {
   
```

It looks like a couple of issues
- It expects `SSHConfig` to be a class, whereas this package exports a 'customised data value' or something (whatever it is, TypeScript doesn't like using it as a type).
- `Directive`, `Line`, and `Section` are not exported


To address this, I would consider making `parse` and `stringify` static methods on the `SSHConfig`, which can be exported directly. Also, as they are reachable through params and return values, the `Line`, `Match`, `Section`, `Directive`, `Comment`, `FindOptions`, and `MatchOptions` interfaces should be exported. `MatchParams` and `ComputeContext` specifically should not be exported, as they are for internal `.compute` logic.

--- a/src/ssh-config.ts
+++ b/src/ssh-config.ts
@@ -10,16 +10,16 @@ const RE_MULTI_VALUE_DIRECTIVE = /^(GlobalKnownHostsFile|Host|IPQoS|SendEnv|User
 const RE_QUOTE_DIRECTIVE = /^(?:CertificateFile|IdentityFile|IdentityAgent|User)$/i
 const RE_SINGLE_LINE_DIRECTIVE = /^(Include|IdentityFile)$/i
 
-enum LineType {
+export enum LineType {
   DIRECTIVE = 1,
   COMMENT = 2,
 }
 
-type Separator = ' ' | '=' | '\t';
+export type Separator = ' ' | '=' | '\t';
 
 type Space = ' ' | '\t' | '\n';
 
-interface Directive {
+export interface Directive {
   type: LineType.DIRECTIVE;
   before: string;
   after: string;
@@ -29,28 +29,28 @@ interface Directive {
   quoted?: boolean;
 }
 
-interface Section extends Directive {
+export interface Section extends Directive {
   config: SSHConfig;
 }
 
-interface Match extends Section {
+export interface Match extends Section {
   criteria: Record<string, string | string[]>
 }
 
-interface Comment {
+export interface Comment {
   type: LineType.COMMENT;
   before: string;
   after: string;
   content: string;
 }
 
-type Line = Match | Section | Directive | Comment;
+export type Line = Match | Section | Directive | Comment;
 
-interface FindOptions {
+export interface FindOptions {
   Host?: string;
 }
 
-interface MatchOptions {
+export interface MatchOptions {
   Host: string;
   User?: string;
 }
@@ -133,9 +133,23 @@ function match(criteria: Match['criteria'], context: ComputeContext): boolean {
   return true
 }
 
-class SSHConfig extends Array<Line> {
+export default class SSHConfig extends Array<Line> {
   static readonly DIRECTIVE: LineType.DIRECTIVE = LineType.DIRECTIVE
   static readonly COMMENT: LineType.COMMENT = LineType.COMMENT
+
+  /**
+   * Parse SSH config text into structured object.
+   */
+  public static parse(text: string): SSHConfig {
+    return parse(text)
+  }
+
+  /**
+   * Stringify structured object into SSH config text.
+   */
+  public static stringify(config: SSHConfig): string {
+    return stringify(config)
+  }
 
   /**
    * Query SSH config by host.
@@ -653,5 +667,3 @@ export function stringify(config: SSHConfig): string {
 
   return str
 }
-
-export default Object.assign(SSHConfig, { parse, stringify })

